pkg/cli/client.go

```diff
@@ -6,32 +6,84 @@ import (
 	"bytes"
 	"context"
 	"crypto/tls"
+	"crypto/x509"
 	"encoding/json"
 	"errors"
 	"io/ioutil"
 	"net/http"
 	"net/url"
+	"os"
+	"path/filepath"
 	"strings"
 	"sync"
 	"time"
 
 	zotErrors "github.com/anuvu/zot/errors"
 )
 
-var httpClient *http.Client //nolint: gochecknoglobals
+var httpClientsMap = make(map[string]*http.Client) //nolint: gochecknoglobals
 
-const httpTimeout = 5 * time.Minute
+const (
+	httpTimeout        = 5 * time.Minute
+	certsPath          = "/etc/containers/certs.d"
+	homeCertsDir       = ".config/containers/certs.d"
+	clientCertFilename = "client.cert"
+	clientKeyFilename  = "client.key"
+	caCertFilename     = "ca.crt"
+)
+
+func createHTTPClient(verifyTLS bool, host string) (*http.Client, error) {
+	// Add a copy of the system cert pool
+	caCertPool, err := x509.SystemCertPool()
+	if err != nil {
+		return nil, err
+	}
 
-func createHTTPClient(verifyTLS bool) *http.Client {
-	var tr = http.DefaultTransport.(*http.Transport).Clone()
+	var insecureSkipVerify bool = false
 	if !verifyTLS {
-		tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: true} //nolint: gosec
+		insecureSkipVerify = true
+	}
+
+	tlsConfig := &tls.Config{
+		RootCAs:            caCertPool,
+		InsecureSkipVerify: insecureSkipVerify, //nolint: gosec
+	}
+
+	// Check if the /home/user/.config/containers/certs.d/$IP:$PORT dir exists
+	home := os.Getenv("HOME")
+	clientCertsDir := filepath.Join(home, homeCertsDir, host)
+
+	if dirExists(clientCertsDir) {
+		returnedTLSConfig, err := getTLSConfig(clientCertsDir, caCertPool, tlsConfig)
+		if err != nil {
+			return nil, err
+		}
+
+		tlsConfig = returnedTLSConfig
+	} else {
+		// Check if the /etc/containers/certs.d/$IP:$PORT dir exists
+		clientCertsDir := filepath.Join(certsPath, host)
+		if dirExists(clientCertsDir) {
+			returnedTLSConfig, err := getTLSConfig(clientCertsDir, caCertPool, tlsConfig)
+			if err != nil && !os.IsPermission(err) {
```

Comment:
Why treat the permissions error differently?
I suggest at least logging permission errors.